src/objects/Term_Grade__c.object

```diff
@@ -0,0 +1,233 @@
+        <actionName>Accept</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>CancelEdit</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>Clone</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>Delete</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>Edit</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>List</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>New</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>SaveEdit</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>Tab</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <actionOverrides>
+        <actionName>View</actionName>
+        <type>Default</type>
+    </actionOverrides>
+    <allowInChatterGroups>false</allowInChatterGroups>
+    <compactLayoutAssignment>SYSTEM</compactLayoutAssignment>
+    <deploymentStatus>Deployed</deploymentStatus>
+    <description>Stores student grades for a specific grading period within a Term.</description>
+    <enableActivities>true</enableActivities>
+    <enableBulkApi>true</enableBulkApi>
+    <enableFeeds>false</enableFeeds>
+    <enableHistory>false</enableHistory>
+    <enableLicensing>false</enableLicensing>
+    <enableReports>true</enableReports>
+    <enableSearch>true</enableSearch>
+    <enableSharing>true</enableSharing>
+    <enableStreamingApi>true</enableStreamingApi>
+    <fields>
+        <fullName>Attempted_Credits__c</fullName>
+        <description>The number of credits attempted for this course and grading period.</description>
+        <externalId>false</externalId>
+        <inlineHelpText>The number of credits attempted for this course and grading period.</inlineHelpText>
+        <label>Attempted Credits</label>
+        <precision>18</precision>
+        <required>false</required>
+        <scale>0</scale>
+        <trackTrending>false</trackTrending>
+        <type>Number</type>
+        <unique>false</unique>
+    </fields>
+    <fields>
+        <fullName>Course_Connection__c</fullName>
+        <deleteConstraint>SetNull</deleteConstraint>
+        <description>The class in which the student is received this grade.</description>
+        <externalId>false</externalId>
+        <inlineHelpText>The class in which the student is received this grade.</inlineHelpText>
+        <label>Course Connection</label>
+        <referenceTo>Course_Enrollment__c</referenceTo>
+        <relationshipLabel>Term Grades</relationshipLabel>
+        <relationshipName>Term_Grades</relationshipName>
+        <required>false</required>
```

Comment:
Just checking - this isn't required, is that intentional? This is currently the only way that a student is tied to a grade, unless we want to make this flexible such that a customer can add a Contact lookup or something if that's desired instead.

Reply:
Would need this explained a bit more to weigh in. What's going on here exactly?

Reply:
Course Connection isn't set as a required field, so someone could create a Term Grade record for nobody, essentially.

We can probably discuss this tomorrow when discussing Attendance, as I think similar questions also apply there.

Reply:
Gotcha. I bet I can conjure up a few use-cases where a  grade needs to be affiliated with a student (contact) without necessarily requiring a Course... like maybe there was a study abroad program and you get one final grade for it, and it's a program (Account), so we should tread lightly here. Agreed we can discuss Friday.

Reply:
I talked to @lifewithryan about this and he suggested that maybe it is made required on the page layout.

Reply:
@xinnanlisf I'm not sure if you saw my last comment here but we should discuss with @aligeez & @lifewithryan if this should be required on the page layout.

Reply:
I think since we decided today that we are going to add the Contact here in a new WI, that it's okay if this is not required.